pygmo/__init__.py

```diff
@@ -62,6 +62,205 @@ class thread_safety(object):
+        # original init.
+        prob_arg = prob
+    else:
+        # Otherwise, we attempt to create a problem from it. This will
+        # work if prob is an exposed C++ problem or a Python UDP.
+        prob_arg = problem(prob)
+    __original_decompose_init(self, prob_arg, weight, z, method, adapt_ideal)
+
+setattr(decompose, "__init__", _decompose_init)
+
+# Override of the unconstrain meta-problem constructor.
+__original_unconstrain_init = unconstrain.__init__
+
+# NOTE: the idea of having the unconstrain init here instead of exposed from C++ is to allow the use
+# of the syntax unconstrain(udp, ... ) for all udps
+def _unconstrain_init(self, prob = None, method = "death penalty", weights = []):
+    """
+    The constructor admits two forms,
+
+    * no arguments,
+    * two mandatory arguments and one optional argument.
+
+    Any other combination of arguments will raise an exception.
+
+    Args:
+        prob: a user-defined problem (either C++ or Python - note that *udp* will be deep-copied
+              and stored inside the :class:`~pygmo.core.unconstrained` instance)
+        method (``str``): a string containing the unconstrain method chosen, one of [``'death penalty'``, ``'kuri'``, ``'weighted'``, ``'ignore_c'``, ``'ignore_o'``]
+        weights (array-like object): the vector of weights to be used if the method chosen is "weighted"
+
+    Raises:
+        ValueError: if either:
+
+        * *prob* is unconstrained,
+        * *method* is not one of [``'death penalty'``, ``'kuri'``, ``'weighted'``, ``'ignore_c'``, ``'ignore_o'``],
+        * *weight* is not of the same size as the problem constraints (if the method ``'weighted'`` is selcted), or not empty otherwise.
+
+        unspecified: any exception thrown by:
+
+        * the constructor of :class:`pygmo.core.problem`,
+        * the constructor of the underlying C++ class,
+        * failures at the intersection between C++ and Python (e.g., type conversion errors, mismatched function
+            signatures, etc.)
+
+    """
+    if prob is None:
+        # Use the null problem for default init.
+        prob = null_problem(nobj=2, nec=3, nic=4)
+    if type(prob) == problem:
+        # If prob is a pygmo problem, we will pass it as-is to the
+        # original init.
+        prob_arg = prob
+    else:
+        # Otherwise, we attempt to create a problem from it. This will
+        # work if prob is an exposed C++ problem or a Python UDP.
+        prob_arg = problem(prob)
+    __original_unconstrain_init(self, prob_arg, method, weights)
+
+setattr(unconstrain, "__init__", _unconstrain_init)
+
+# Override of the mbh meta-algorithm constructor.
+__original_mbh_init = mbh.__init__
+# NOTE: the idea of having the mbh init here instead of exposed from C++ is to allow the use
+# of the syntax mbh(uda, ...) for all udas
+def _mbh_init(self, algo = None, stop = 5, perturb = 1e-2, seed = None):
+    """
+    The constructor admits two forms,
+
+    * no arguments,
```

Comment:
I removed them all. The problem is that is the user does call the constructor with the wrong form, there may be an exception in c++. For example if you call translate with a kwarg translate=[1.1,1.2] since the default value for the problem is null_problem and that has dimension 1 ....

Anyway as I said I removed all it will be up to the user to figure what was the intended use.